src/main/java/ch/njol/skript/expressions/ExprWorld.java

```diff
@@ -109,10 +109,13 @@ public boolean setTime(final int time) {
 	
 	@Override
 	public void change(Event e, @Nullable Object[] delta, Changer.ChangeMode mode) {
-		if (!(getExpr().getAll(e) instanceof Location[] && delta != null)) return;
+		Object[] objects = getExpr().getAll(e);
```

Comment:
Change it to getArray while you're at it. This is old code so I suppose it wasn't the standard.